### PR TITLE
Build authentication

### PR DESCRIPTION
**Pull request by harethriyadh**

### CODE REVIEW

File: src/routes/index.js
Comment (motaz99) [inline src/routes/index.js:1]: If you delete this file, the app won't work, because you are importing it in the app.js. So why did you delete it?
Try to run your app, it won't work.

File: yarn.lock
Comment (motaz99) [inline yarn.lock:1]: This file have a lot of changes, if you can see that 4789 lines got modified, and I think this is be cause either we have different version of node, or a different version of yarn. 
my yarn version is 1.22.19
my node version is 16.13.1

So what are you going to to do is:
1. Check your node version and if it not like the my version then you need to make it like that. it's not nessessarly to be the exact version of my but for example my node 16.something, you need to install 16 and any sub version of that is ok
2. You are going to do the same for the yarn, so you need to have 1.22.19, for yarn try to use the exact same one.

File: src/models/auth.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    required: true,
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  role: {
+    type: String,
+    enum: ['admin', 'user'],
```
Comment (motaz99) [inline src/models/auth.js]: We don't have admin as a role, what is admin?
The `enum` means one of those in the array right? so we will have either `chef` or `client` I don't know where from did you got the admin from.

File: src/models/index.js
Comment (motaz99) [inline src/models/index.js:1]: why did you delete this file? This is not your task.
Your task is to work authentication.
Get it back please.

File: src/models/auth.js
Comment (motaz99) [inline src/models/auth.js:1]: We need to change the name of this file later, because there is not modal named auth, modal is something that indicate to the schema, and even your schema named `loginSchema` but it actually a User schema.

We will keep it for now but this need to get refactor later.

File: src/jwt.js
Comment (motaz99) [inline src/jwt.js:1]: We don't need the comments in this file, delete them, please.

File: src/controllers/index.js
Comment (motaz99) [inline src/controllers/index.js:1]: Your task to work on the authentication not deleting files. get it back, please.

File: src/controllers/authController.js
Comment (motaz99) [inline src/controllers/authController.js:1]: As a general comment on this file, we don't need unnecessary comments, for example on line 12 you wrote // checking if the user exists, but actually, we don't need this, I can understand that by reading the code.

File: src/controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    const isPasswordValid = await bcrypt.compare(password, user.password);
+    if (!isPasswordValid) {
+      return res.status(401).json({ error: 'Invalid  email or password' });
+    }
+
+    // create JWT access token
+    const accessToken = createTokens(user);
+
+    // save the JWT token to the cookies
+    res.cookie('access-token', accessToken, {
```
Comment (motaz99) [inline src/controllers/authController.js]: We need to change the name of the cookie to 'jwt` instead of `access-token`

File: src/controllers/authController.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    await user.save();
+
+    // Password reset successful
+    res.json({ message: 'Password reset successful' });
+
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+};
```
Comment (motaz99) [inline src/controllers/authController.js]: We talked about this, and we decide to leave it for now. do you remember?

File: src/controllers/authController.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+// endpoint to check if user is authorized or not
+const profile = (req, res) => {
+  console.log(req.userId);
+  res.json('profile');
+};
```
Comment (motaz99) [inline src/controllers/authController.js]: how does this work?

File: src/controllers/authController.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+// endpoint to check if user is authorized or not
+const dashboard = (req, res) => {
+  res.json('dashboard');
+};
```
Comment (motaz99) [inline src/controllers/authController.js]: How this endpoint will check if the user authorized or not?

**Review by motaz99 — CHANGES_REQUESTED** [review]
Currently the app will not work, and this is because the index.js file inside the route folder got deleted, we used that file as a bridge between the app.js and the authRoutes.js that you have created.

I already mentioned that, you are app now is not working. 

What you need to do now is:
1. Go through all of the comments that I did in this PR
2. Do the needed changes
3. You need to push your changes into the same branch
4. And you can ask me to redo the review from the same place that you asked to review your code for the first time.

File: yarn.lock
Comment (harethriyadh) [reply yarn.lock:1]: my yarn  -v > v1.22.19
my node -v > v16.13.1

File: src/controllers/authController.js
Comment (harethriyadh) [reply src/controllers/authController.js:1]: .

File: src/controllers/authController.js (hunk 2 above)
Comment (harethriyadh) [reply src/controllers/authController.js]: why?

File: src/controllers/authController.js (hunk 4 above)
Comment (harethriyadh) [reply src/controllers/authController.js]: In summary, the /dashboard route first validates the token using the validateToken middleware and then checks if the user is an admin using the isAdmin middleware. Only if both conditions are met, the request reaches the dashboard function to send the 'dashboard' JSON response. Otherwise, an appropriate error response is sent.

File: src/controllers/authController.js (hunk 5 above)
Comment (harethriyadh) [reply src/controllers/authController.js]: as an example of an authorized process, I used isAdmin() function with endpoint /dashboard.
we can add more authorization checking later on when we will do the other endpoints.

File: src/jwt.js
Comment (harethriyadh) [reply src/jwt.js:1]: someone else will need it.

File: src/models/auth.js
Comment (harethriyadh) [reply src/models/auth.js:1]: we did not make any deal about file naming so I did it in free logic.

File: src/models/auth.js (hunk 1 above)
Comment (harethriyadh) [reply src/models/auth.js]: .

Comment (motaz99) [reply src/models/auth.js]: what does this . mean?

File: src/models/auth.js
Comment (motaz99) [reply src/models/auth.js:1]: Harith you don't to replay on this, I told you in the comment that we will keep it for now, I just wrote it so keep it in mind.

File: src/controllers/authController.js
Comment (motaz99) [reply src/controllers/authController.js:1]: Again what does the . mean?

File: src/controllers/authController.js (hunk 4 above)
Comment (motaz99) [reply src/controllers/authController.js]: Again we don't nave admin

File: src/controllers/authController.js (hunk 5 above)
Comment (motaz99) [reply src/controllers/authController.js]: We don't have admin.

**Comment by motaz99** [comment]
close #15 
close #16
